src/hooks/common/useBottomSheet.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState, useCallback, useRef, useEffect } from 'react';
+
+import { DragInfo } from '@/components/common/BottomSheet/BottomSheet.types';
+import { INITIAL_HEIGHT, MIN_VISIBLE_HEIGHT, MAX_HEIGHT } from '@/constants/bottomSheetOptions';
+
+export const useBottomSheet = (resetTrigger: boolean, onClose?: () => void) => {
+  const [sheetHeight, setSheetHeight] = useState(INITIAL_HEIGHT);
+  const [isHidden, setIsHidden] = useState(false);
+  const [isInteractionDisabled, setIsInteractionDisabled] = useState(false);
+  const dragOffsetRef = useRef(0);
+  const initialPositionRef = useRef(INITIAL_HEIGHT);
+
+  useEffect(() => {
+    if (resetTrigger) {
+      setSheetHeight(INITIAL_HEIGHT);
+      setIsHidden(false);
+      setIsInteractionDisabled(false);
+      dragOffsetRef.current = 0;
+      initialPositionRef.current = INITIAL_HEIGHT;
+    }
+  }, [resetTrigger]);
+
+  const handleDrag = useCallback(
+    (_: MouseEvent | TouchEvent | PointerEvent, info: DragInfo) => {
+      if (isInteractionDisabled) return;
+
+      requestAnimationFrame(() => {
+        const dragAmount = -info.delta.y;
+        dragOffsetRef.current += dragAmount;
+
+        const newHeight = Math.min(
+          Math.max(initialPositionRef.current + dragOffsetRef.current, MIN_VISIBLE_HEIGHT),
+          MAX_HEIGHT
+        );
+        setSheetHeight(newHeight);
+      });
+    },
+    [isInteractionDisabled, setSheetHeight]
+  );
+
+  const handleDragEnd = useCallback(() => {
+    if (sheetHeight <= MIN_VISIBLE_HEIGHT) {
+      setIsHidden(true);
+      setIsInteractionDisabled(true);
+      if (onClose) {
+        onClose();
+      }
+    } else if (sheetHeight > MAX_HEIGHT) {
+      setSheetHeight(MAX_HEIGHT);
+    }
+  }, [sheetHeight, onClose]);
```

Comment:
둘 중 어느게 더 가독성이 좋으신가요?? `early return` 을 사용하면 가독성이 올라갑니다! 참고해서 읽어보면 좋을 것 같아요.

[Tidy First?](https://velog.io/@bo-like-chicken/tidy-first-%EC%BD%94%EB%93%9C-%EC%A0%95%EB%A6%AC%EB%B2%95)

```suggestion
  const handleDragEnd = useCallback(() => {
    if (onClose) {
      return onClose();
     }
    if (sheetHeight <= MIN_VISIBLE_HEIGHT) {
      setIsHidden(true);
      setIsInteractionDisabled(true);
      return;
    }
     if (sheetHeight > MAX_HEIGHT) {
      setSheetHeight(MAX_HEIGHT);
      return;
    }
  }, [sheetHeight, onClose]);
```

Reply:
옷 알려주셔서 감사합니다 훨씬 읽기 편합니다 !!